src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,445 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.7.1;
+pragma experimental ABIEncoderV2;
+
+import "../common/BaseWithStorage/ERC721BaseToken.sol";
+import "../common/Interfaces/AssetToken.sol";
+import "../common/Interfaces/IGameToken.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+contract GameToken is ERC721BaseToken, IGameToken {
+    ///////////////////////////////  Libs //////////////////////////////
+
+    using SafeMath for uint256;
+
+    ///////////////////////////////  Data //////////////////////////////
+
+    // contract responsible for forwarding specific calls to GameToken contract.
+    address internal _gameManager;
+    AssetToken _asset;
+
+    bytes4 private constant ERC1155_RECEIVED = 0xf23a6e61;
+    bytes4 private constant ERC1155_BATCH_RECEIVED = 0xbc197c81;
+    uint256 private constant CREATOR_OFFSET_MULTIPLIER = uint256(2)**(256 - 160);
+
+    mapping(uint256 => mapping(uint256 => uint256)) private _gameAssets;
+    mapping(address => address) private _creatorship; // creatorship transfer
+
+    mapping(uint256 => string) private _metaData;
+    mapping(uint256 => mapping(address => bool)) private _gameEditors;
+
+    ///////////////////////////////  Events //////////////////////////////
+
+    event Minter(address newMinter);
```

Comment:
we are using the word gameManager and minter.
I guess we should pick one.
if we use Minter, we could reuse the WithMinter contract: https://github.com/thesandboxgame/sandbox-smart-contracts/blob/hardhat/src/solc_0.7/common/BaseWithStorage/WithMinter.sol

Reply:
I think the ability to reuse WithMinter is a good reason to use Minter over GameManager.